File: Projects/10/Ruby/Penguin.rb
# Penguin.rb | Defines the Penguin class which inherits attributes and methods
#   from the Walkingbird superclass.
#
# Completed by: Emily Costa
# Date: 4/27/2020
####################################################

require './Walkingbird.rb'

class Penguin < Walkingbird

    def initialize(name)
        @name = name
    end

    def className
        "Penguin"
    end

    def call
        "Huh-huh-huh!"
    end

end

File: Projects/10/Java/Penguin.java
/* Penguin.java provides a Penguin subclass of Flyingbird.
 *
 * Completed by: Emily Costa
 * Date: 4/27/2020
 ******************************************************/

public class Penguin extends Walkingbird{
    
    /* explicit constructor
     * Receive: name, a String 
     * PostCond: myName == name.
     */
    public Penguin(String name) {
        super(name);
    }

    /* A Penguin's class name
     * Return: a Penguin's class name
     */
    public String getClassName() {
        return "Penguin";
    }
   
    /* A Penguin's Call
     * Return: a penguin-call ("huh-huh-huh!").
     */
    public String call() {
        return "Huh-huh-huh!";
    }
   
   }
   
   